crates/ironrdp-async/src/connector.rs

```diff
@@ -187,10 +187,23 @@ where
     S: FramedWrite + FramedRead,
 {
     buf.clear();
+    let written = single_connect_step_read(framed, connector, buf).await?;
+    single_connect_step_write(framed, buf, written).await
+}
+
+pub async fn single_connect_step_read<S>(
```

Comment:
question: Is it intended to be public?

Reply:
It's used by Teleport for executing the deactivation-reactivation sequence. It appears to be a generally useful pattern, since I also needed it the [attempted] implementation for IronRDP web: https://github.com/Devolutions/IronRDP/pull/421/files#diff-33dc7055fcffcb2318664572070611527a75668ad24c01e2ce01ae83e0b27defR616